### ISSUE TITLE
Cumbersome to work test driven when testing source from pods while using `use_frameworks!`

### ISSUE THREAD

**Issue by hfossli** [CocoaPods/CocoaPods#5024]
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
Activated `use_frameworks!`

## What did you expected to happen?
Without `use_frameworks!` I was able to have a local dependency from my demo project to the source whilst any changes I made to header files were immediately available in unit test. 

## What happened instead?
With `use_frameworks!` I now have to compile in order for auto complete to work. This is a tedious process and keeps me from shifting to activate `use_frameworks` on many projects. 

[See this video](https://github.com/CocoaPods/CocoaPods/files/168915/video.zip). In the video you will see that only after I build the new functions are available with auto complete in the unit tests.

## Podfile
https://github.com/hfossli/AGGeometryKit/blob/master/Demo/Podfile
```ruby
use_frameworks!

pod 'AGGeometryKit', :path => '../AGGeometryKit.podspec'

target :AGGeometryKitDemoTests do
  pod 'Expecta', :inhibit_warnings => true
end
```

## Setup
Xcode Version 7.2 (7C68)
Cocoapods 0.39.0

## Discussion
I think this problem is not related to Cocoapods directly as the change is merely because we're using modules instead of plain headers. I do think this is a problem for the community and I would like to know what workarounds other people are using.

## Possible workaround
If I/you want to unit test or work test driven, don't depend to source code available through modules/cocoapods. Is this what everyone else does?

**Comment by danielgalasko** [comment]
Could also try and post this on the Apple dev forums or file a radar... I haven't really had issues with this in the 6 pods we use internally

**Comment by hfossli** [comment]
@danielgalasko do you have the same setup? And do you unit test the source of those pods?

**Comment by danielgalasko** [comment]
Similar setup yes, and yes every pod has its own set of tests etc

```
use_frameworks!

target 'SpreeAPI_Example' do
    pod 'SpreeAPI', :path => "../"
    target 'SpreeAPI_Tests' do
        inherit! :search_paths
        pod 'OHHTTPStubs'
    end
end
```

I have recently switched to pods 1.0.0 but I doubt that makes a difference. Also on Xcode 7 etc which I am sure you are on too. Have you tried creating a new repo `pod lib create` and going down that route to see? Might be a particular issue with your specific pod...

**Comment by segiddins** [comment]
Why is building an issue?

**Comment by hfossli** [comment]
Normally it is sufficient to just hit cmd+s and then jump over to the unit test and type away. The delay involved with having to wait the extra seconds it takes for build to complete is what is making this an issue for me.

**Comment by segiddins** [comment]
Can you check if 1.0 fixes this? Since it involves copying fewer headers it might. Also, verifying that this isn't an issue with Xcode itself would be appreciated.

**Comment by mrackwitz** [comment]
Are you using module imports or header imports? The former might have a shorter / more reliable path for Xcode to discover implicit dependencies. You could try out adding the `Pods.xcodeproj` as subproject to your project and adding the dependencies to your test target as explicit dependencies in Xcode.

**Comment by hfossli** [comment]
I will get back to you on Monday or Tuesday.

**Comment by mgrebenets** [comment]
This should be related to #4944 and I've prepared a demo project that demonstrates similar issue: https://github.com/mgrebenets/cocoapods-tests-with-transient-binary

**Comment by hfossli** [comment]
This project, _Poddy_, was set up with `pod lib create`

It is necessary to recompile before autocomplete kicks in when
- changing api in objc and testing from objc
- changing api in swift and testing from objc
- changing api in objc and testing from swift
- changing api in swift and testing from swift

------

This project, _Direct_, is just a plain xcode project – same as Poddy, but with source embedded directly instead of integrated via cocoapods / modules.

It is necessary to build **unit test target** before autocomplete kicks in when
- changing api in swift and testing from objc

This seems to be because I had to add Moddy.swift to the test target manually. I'm not sure why I had to do that...

----- 

Let me know if my findings here are incorrect. I am a little bit surprised. Anyways, it seems to be better having the source of the pod you are creating embedded directly in xcode and not integrated through pods. This might have implications for how the base template for `pod lib create` should be. 

Recompilation is not a big problem unless it takes some time. I will do more testing to check wether this actually is an issue or not. 

[Direct.zip](https://github.com/CocoaPods/CocoaPods/files/176053/Direct.zip)
[Poddy.zip](https://github.com/CocoaPods/CocoaPods/files/176052/Poddy.zip)
